hoodie-client/src/main/java/com/uber/hoodie/table/HoodieMergeOnReadTable.java

```diff
@@ -203,14 +207,40 @@ public HoodieCompactionPlan scheduleCompaction(JavaSparkContext jsc, String inst
             return false;
           };
 
+          final Map<FileStatus, Boolean> filesToDeletedStatus = new HashMap<>();
+
           switch (instant.getAction()) {
             case HoodieTimeline.COMMIT_ACTION:
-            case HoodieTimeline.COMPACTION_ACTION:
               try {
-                Map<FileStatus, Boolean> results = super
-                    .deleteCleanedFiles(partitionPath, Collections.singletonList(commit));
+                // Rollback of a commit should delete the newly created parquet files along with any log
+                // files created with this as baseCommit. This is required to support multi-rollbacks in a MOR table.
+                super.deleteCleanedFiles(filesToDeletedStatus, partitionPath, filter);
                 hoodieRollbackStats = HoodieRollbackStat.newBuilder()
-                    .withPartitionPath(partitionPath).withDeletedFileResults(results).build();
+                    .withPartitionPath(partitionPath).withDeletedFileResults(filesToDeletedStatus).build();
+                break;
+              } catch (IOException io) {
+                throw new UncheckedIOException("Failed to rollback for commit " + commit, io);
+              }
+            case HoodieTimeline.COMPACTION_ACTION:
+              try {
+                boolean higherDeltaCommits = !this.getActiveTimeline().reload().getDeltaCommitTimeline()
+                    .findInstantsAfter(commit, 1).empty();
+                if (higherDeltaCommits) {
+                  // Rollback of a compaction action means that the compaction is scheduled and has not yet finished.
+                  // In this scenario we should delete only the newly created parquet files and not corresponding base
+                  // commit log files created with this as baseCommit since updates would have been written to the log
+                  // files.
+                  super.deleteCleanedFiles(filesToDeletedStatus, commits, partitionPath);
+                  hoodieRollbackStats = HoodieRollbackStat.newBuilder()
+                      .withPartitionPath(partitionPath).withDeletedFileResults(filesToDeletedStatus).build();
+                } else {
+                  // No deltacommits present after this compaction commit (inflight or requested). In this case, we
+                  // can also delete any log files that were created with this compaction commit as base
+                  // commit.
+                  super.deleteCleanedFiles(filesToDeletedStatus, partitionPath, filter);
+                  hoodieRollbackStats = HoodieRollbackStat.newBuilder()
```

Comment:
Since, this is one really complex and important functionalities of Hoodie. Can we add lot of Pre/Post Condition checks. One condition, I can think of is to

(a) Reload timeline. Check if the timeline do not show the rolled-back instant.
(b) Recreate the file system view for all the partitions that were loaded previously. Check if the latest file-slice does not have the commit timestamp as rollback timestamp

Reply:
I'm not adding these checks as they make the code more complicated, we can do such checks in general in the TableFileSystemView as we discussed. Also, I changed the way inflight files are deleted, please take a pass.

Reply:
Yes, We ensure for safety in HoodieFileGroup.getAllFileSlices()